src/Layout/QueryParamContext.tsx

```diff
@@ -219,8 +220,21 @@ export default function QueryParamProvider({ children }: any) {
     }
   }, [paymentError]);
   React.useEffect(() => {
-    if (router.query.locale) {
+    if (allLocales.find((locale) => locale.key === router.query.locale)) {
       setlanguage(router.query.locale);
+    } else {
+      //conditional operator to check if navigator.languages property is supported by browser.
+      const userLocale =
+        navigator.languages === undefined
+          ? [navigator.language]
+          : navigator.languages;
+      const newLocale = userLocale[0].trim().split(/-|_/)[0];
+
+      if (allLocales.find((locale) => locale.key === newLocale)) {
+        setlanguage("en");
```

Comment:
If above condition resolves to `true` shouldn't we set newLocale here?

Reply:
Makes sense, thanks!